### PR TITLE
PTTB-180 Admin delete protected

### PR DESCRIPTION
**Pull request by Rznjoc**

### CODE REVIEW

File: app/Http/Requests/UserStoreRequest.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ public function rules()
         return [
             'name' => 'required|regex:/^[\pL\s\-]+$/u|max:255',
             'email' => 'required | email|max:255',
-            'password' => ['required','min:6','regex:/^.*(?=.{3,})(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[\d\x])(?=.*[!$#%]).*$/','confirmed'],
+            'password' => ['required','min:6','regex:/^.*(?=.{3,})(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[\d\x])(?=.*[!@#$%^&*]).*$/','confirmed'],
```
Comment (Prabeg29) [inline app/Http/Requests/UserStoreRequest.php:29]: @Rznjoc Why not use the simple `Password` rule Laravel provides than to use this complex regex

File: app/Http/Controllers/AdminController.php [hunk 2]
```diff
@@ -12,22 +12,40 @@
...
+        }
+
         $user->delete();
+        
         return response()->json([
-            'message' => 'User deleted successfully'
+            'message' => 'User deleted Successfully'
         ], 200);
+
+        return response()->json([
```
Comment (Prabeg29) [inline app/Http/Controllers/AdminController.php:46]: @Rznjoc multiple return statements in the same block?

File: routes/api.php [hunk 3]
```diff
@@ -67,7 +67,6 @@
             Route::post('/', 'addActivity');
             Route::patch('{id}', 'updateActivity');
             Route::patch('project-status/{id}', 'updateProjectStatus');
-            Route::delete('{id}', 'deleteProject');
```
Comment (Prabeg29) [inline routes/api.php:70]: @Rznjoc Why is this route deleted? There is still functionality in FE to delete project

File: app/Services/AdminService.php [hunk 4]
```diff
@@ -2,15 +2,20 @@
...
 namespace App\Services;
 
-use App\Models\UserRole;
+use App\Models\User;
+use Illuminate\Http\Response;
 
 class AdminService
 {
-    public static function deleteRoles($id)
+    public static function deleteUser($id)
```
Comment (Prabeg29) [inline app/Services/AdminService.php:10]: @Rznjoc This action according to naming does not match the functionality

**Review by Prabeg29 — CHANGES_REQUESTED** [review]

File: app/Http/Requests/UserStoreRequest.php (hunk 1 above)
Comment (Rznjoc) [reply app/Http/Requests/UserStoreRequest.php:29]: As FE has used Regex so this is done

**Review by Rznjoc — COMMENTED** [review]
deleteProject route added

File: app/Http/Requests/UserStoreRequest.php (hunk 1 above)
Comment (Prabeg29) [reply app/Http/Requests/UserStoreRequest.php:29]: @Rznjoc Brother check the Password rules and compare them with this. If you a real OG and prefer it the way it is, fine.

File: app/Http/Controllers/AdminController.php (hunk 2 above)
Comment (Rznjoc) [reply app/Http/Controllers/AdminController.php:47]: done

File: app/Services/AdminService.php (hunk 4 above)
Comment (Rznjoc) [reply app/Services/AdminService.php:10]: is it not OK for now?

File: app/Http/Requests/UserStoreRequest.php (hunk 1 above)
Comment (Rznjoc) [reply app/Http/Requests/UserStoreRequest.php]: FE said this will be fine for now brother.

**Review by Rznjoc — COMMENTED** [review]
Final Changes

File: app/Services/UserService.php [hunk 5]
```diff
@@ -82,12 +82,4 @@ public static function resetPassword(array $validatedResetPass): bool
         return true;
     }
 
-    public static function checkUserIdExists($id)
```
Comment (sakaruk) [inline app/Services/UserService.php:85]: Merge With the master and check if the function is used in other places too. I see that timelog controller uses it currently. Before removing function please check the dependencies

**Review by sakaruk — CHANGES_REQUESTED** [review]

File: app/Services/UserService.php (hunk 5 above)
Comment (Rznjoc) [reply app/Services/UserService.php]: Okay dai i will keep this in mind :)

File: app/Services/AdminService.php [hunk 6]
```diff
@@ -2,15 +2,20 @@
...
 namespace App\Services;
 
-use App\Models\UserRole;
+use App\Models\User;
+use Illuminate\Http\Response;
 
 class AdminService
 {
-    public static function deleteRoles($id)
+    public static function checkUserIdExists($id)
```
Comment (sakaruk) [inline app/Services/AdminService.php:10]: Isn't this the same function that is in UserService ? Is it something different ?

**Review by sakaruk — CHANGES_REQUESTED** [review]

**Review by sakaruk — APPROVED** [review]

File: app/Services/AdminService.php (hunk 4 above)
Comment (Rznjoc) [reply app/Services/AdminService.php]: Function name changed to 'checkUserIdExists' which is already a used function in this project.

**Review by Prabeg29 — APPROVED** [review]

**Pull request merged**